api/application_submissions/index.js

```diff
@@ -31,24 +31,31 @@ router.post('/', async (req, res) => {
   }
 });
 
-router.put('/:id', async (req, res) => {
-  const { id } = req.params;
+router.put('/', async (req, res) => {
+  const { ids } = req.body; 
   const { decision } = req.body;
 
   try {
-    let applicationSubmission = await ApplicationSubmission.findById(id);
-    if (applicationSubmission) {
-      if (decision === 'ACCEPTED') {
-        const { skilled_impact_request_id } = applicationSubmission;
-        applicationSubmission = await ApplicationSubmission.acceptAndDenyAllOthers(id, skilled_impact_request_id);
-      } else {
-        applicationSubmission = await ApplicationSubmission.update(id, decision);
-      }
-      res.status(200).json({ applicationSubmission, message: 'Submission updated in database' });
-    } else {
-      res.status(404).json({ message: 'Submission not found in database' });
+    const applicationSubmissions = await ApplicationSubmission.updateAll(ids, decision);
+    // return 200 if all submissions successfully updated
+    if(applicationSubmissions.length == ids.length) {
+      res.status(200).json({ applicationSubmissions, message: 'Submissions updated in database' });
     }
-  } catch (error) {
+    // return 404 if no submissions found, else return 202 (accepted, but some processing still needed)
+    // for when some submissions are updated and some aren't
+    else if(applicationSubmissions.length == 0){ 
+      res.status(404).json({ message: 'Submissions not found in database' });
+    }
+    else {
+      const numSubmissionsFound = applicationSubmissions.length;
+      const numSubmissionsNotFound = ids.length - numSubmissionsFound;
+      res.status(202).json({
```

Comment:
On second thought, are we *sure* we don't want to fail here? We've received a request regarding data that doesn't exist and it's a sign of something gone wrong. We should also list the IDs that weren't found in the response, not just the number of not-found submissions.

Reply:
Ah, I see what you're saying. So since something may have gone wrong in the client to send even 1 wrong id to the server, we should not update anything at all, correct?

Reply:
Yeah, I would error with a 404 here and list the IDs that weren't found.